crates/svm-cli/tests/app_template_tests.rs

```diff
@@ -0,0 +1,164 @@
+extern crate regex;
+extern crate tempfile;
+
+use regex::Regex;
+use std::fs::File;
+use std::io::Read;
+use svm_cli::cli;
+
+struct AppTemplateTestCase {
+    version: String,
+    name: String,
+    page_count: String,
+}
+
+#[test]
+fn encode_decode() {
+    let cases = vec![
+        AppTemplateTestCase {
+            version: String::from("0"),
+            name: String::from("my name"),
+            page_count: String::from("0"),
+        },
+        AppTemplateTestCase {
+            version: String::from("1"),
+            name: String::from("שלום"),
+            page_count: String::from("1000"),
+        },
+        AppTemplateTestCase {
+            version: String::from("1"),
+            name: String::from("नमस्ते"),
+            page_count: String::from("1000"),
+        },
+    ];
+
+    for case in cases {
+        test_encode_decode(case)
+    }
+}
+
+#[test]
+fn encode_invalid_codepath() {
+    let version = "0";
+    let name = "";
+    let page_count = "1";
+    let code_path = "invalid_codepath";
+    let output_path = "";
+    let input = vec![
+        "myprog",
+        "encode",
+        "app-template",
+        version,
+        name,
+        page_count,
+        code_path,
+        output_path,
+    ];
+    println!("input: {:?}", input);
+
+    let matches = cli::new_app().get_matches_from(input);
+    match cli::process(matches) {
+        Ok(_) => panic!(),
+        Err(e) => assert!(e
+            .to_string()
+            .starts_with(&format!("failed to open file at {}", code_path))),
+    }
+}
+
+#[test]
+fn encode_invalid_outputpath() {
+    let version = "0";
+    let name = "";
+    let page_count = "1";
+    let code_path = &gen_wasm_example_path();
+    let output_path = "";
+    let input = vec![
+        "myprog",
+        "encode",
+        "app-template",
+        version,
+        name,
+        page_count,
+        code_path,
+        output_path,
+    ];
+    println!("input: {:?}", input);
+
+    let matches = cli::new_app().get_matches_from(input);
+    match cli::process(matches) {
+        Ok(_) => panic!(),
+        Err(e) => assert!(e
+            .to_string()
+            .starts_with(&format!("failed to create file at {}", output_path))),
+    }
+}
+
+#[test]
+fn decode_invalid_datapath() {
+    let data_path = "non_existing_path";
+    let input = vec!["myprog", "decode", "app-template", data_path];
+    println!("input: {:?}", input);
+
+    let matches = cli::new_app().get_matches_from(input);
+    match cli::process(matches) {
+        Ok(_) => panic!(),
+        Err(e) => assert!(e
+            .to_string()
+            .starts_with(&format!("failed to open file at {}", data_path))),
+    }
+}
+
+fn test_encode_decode(case: AppTemplateTestCase) {
+    let wasm_example_path = &gen_wasm_example_path();
```

Comment:
I'd personally use a shorter name here

Reply:
I changed it to a plain const instead, because calling to `fs::canonicalize` wasn't necessary. 
Didn't change the name though.